src/scripts/GoogleTagManagerLoader.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @implements {GoogleTagManagerLoader}
+ * @memberof module:GoogleTagManager
+ */
+class GoogleTagManagerLoader {
+  constructor({ config }) {
+    this.gtmId = config.GOOGLE_TAG_MANAGER_ID;
+    this.gtmArgs = '';
+    if (config.GOOGLE_TAG_MANAGER_AUTH) {
+      this.gtmArgs += `&gtm_auth=${config.GOOGLE_TAG_MANAGER_AUTH}`;
+    }
+    if (config.GOOGLE_TAG_MANAGER_PREVIEW) {
+      this.gtmArgs += `&gtm_preview=${config.GOOGLE_TAG_MANAGER_PREVIEW}`;
+    }
+    if (config.GOOGLE_TAG_MANAGER_ADDNL_ARGS) {
+      if (!config.GOOGLE_TAG_MANAGER_ADDNL_ARGS.startsWith('&')) {
+        this.gtmArgs += '&';
+      }
+      this.gtmArgs += config.GOOGLE_TAG_MANAGER_ADDNL_ARGS;
+    }
+  }
+
+  loadScript() {
+    if (!this.gtmId) {
+      return;
+    }
+
+    global.googleTagManager = global.googleTagManager || [];
+    const { googleTagManager } = global;
+
+    // If the snippet was invoked do nothing.
+    if (googleTagManager.invoked) {
```

Comment:
[curious-passerby] Could you give an example where loadScript would be called twice? I feel like it is only being called from initialize, and initialize should only be called once in an app, right?